slackclient/_client.py

```diff
@@ -19,7 +19,18 @@ def rtm_connect(self):
             return False
 
     def api_call(self, method, **kwargs):
-        return self.server.api_call(method, **kwargs)
+        result = json.loads(self.server.api_call(method, **kwargs))
+        if self.server:
+            if method == 'im.open':
+                if "ok" in result and result["ok"]:
+                    self.server.attach_channel(kwargs["user"], result["channel"]["id"], [])
+            elif method in ('mpim.open', 'groups.create', 'groups.createchild'):
+                if "ok" in result and result["ok"]:
+                    self.server.attach_channel(result['group']['name'], result['group']['id'], [])
```

Comment:
I was emulating the style of _client.py:SlackClient:process_changes, which also passes an empty. I'll update it, one moment.